src/function/scalar/map/map_from_entries.cpp

```diff
@@ -0,0 +1,85 @@
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/parser/expression/bound_expression.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/pair.hpp"
+
+namespace duckdb {
+
+static void MapFromEntriesFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(result.GetType().id() == LogicalTypeId::MAP);
+
+	// ! Otherwise if its not a constant vector, this breaks the optimizer
+	result.SetVectorType(VectorType::CONSTANT_VECTOR);
+	for (idx_t i = 0; i < args.ColumnCount(); i++) {
+		if (args.data[i].GetVectorType() != VectorType::CONSTANT_VECTOR) {
+			result.SetVectorType(VectorType::FLAT_VECTOR);
+		}
+	}
+
+	// Get the arguments vector
+	auto &array = args.data[0];
+	idx_t pair_amount = ListVector::GetListSize(array);
+	auto &entries = ListVector::GetEntry(array);
+
+	// Prepare the result vectors
+	result.Resize(0, pair_amount);
+	auto &child_entries = StructVector::GetEntries(result);
+	D_ASSERT(child_entries.size() == 2);
+	auto &key_vector = child_entries[0];
+	auto &value_vector = child_entries[1];
+
+	auto key_data = FlatVector::GetData<list_entry_t>(*key_vector);
+	auto value_data = FlatVector::GetData<list_entry_t>(*value_vector);
+	// Transform to mapped values
+	key_data[0].offset = 0;
+	value_data[0].offset = 0;
+	for (idx_t i = 0; i < pair_amount; i++) {
+		auto element = entries.GetValue(i);
+		D_ASSERT(element.type().id() == LogicalTypeId::STRUCT);
+		auto &key_value = StructValue::GetChildren(element);
+		ListVector::PushBack(*key_vector, key_value[0]);
+		ListVector::PushBack(*value_vector, key_value[1]);
+	}
+	key_data[0].length = pair_amount;
+	value_data[0].length = pair_amount;
+
+	result.Verify(args.size());
+}
+
+static unique_ptr<FunctionData> MapFromEntriesBind(ClientContext &context, ScalarFunction &bound_function,
+                                                   vector<unique_ptr<Expression>> &arguments) {
+	child_list_t<LogicalType> child_types;
+
+	if (arguments.size() != 1) {
+		throw Exception("We need one list of structs for a map");
+	}
+	auto &list = arguments[0]->return_type;
+	if (list.id() != LogicalTypeId::LIST) {
+		throw Exception("Argument is not a list");
```

Comment:
Here, you could use a more specific exception. E.g. `throw InvalidInputException("The input argument must be a list of structs.");` For binder exceptions, there is also a `BinderException(...)`.